src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs

```diff
@@ -14,7 +14,11 @@ public static class ModelDirective
         public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
             "model",
             DirectiveKind.SingleLine,
-            builder => builder.AddTypeToken());
+            builder =>
+            {
+                builder.AddTypeToken();
+                builder.Usage = DirectiveUsage.SinglePreContent;
```

Comment:
Should usage just be a required argument to `CreateDirective`? I don't have a major grip with this, I just want to understand how you see the tradeoffs.